### PR TITLE
[CDE-940] - Persistent Cross-site scripting (XSS) vulnerability in sy…

### PR DESCRIPTION
**Pull request by YuryBY**
…nchoriseDashboard

### CODE REVIEW

**Comment by YuryBY** [comment]
@pamval, could you please take a look?

**Comment by pamval** [comment]
@afrjorge please review.

File: webclient/impls/pentaho/src/main/javascript/cdf-dd-base.js [hunk 1]
```diff
@@ -534,8 +534,8 @@ var SaveRequests = {
     $.post(wd.cde.endpoints.getPluginUrl() + "syncronizer/syncronizeDashboard", saveSettingsParams, function(result) {
       try {
         if(result && result.status == "true") {
-          myself.setDashboardWcdf(wcdf);
-          refreshTitle(wcdf.title);
+          myself.setDashboardWcdf(result.result.wcdf);
```
Comment (pamval) [inline webclient/impls/pentaho/src/main/javascript/cdf-dd-base.js]: do we really need to use result.result? can't we  have just one result?

File: core/impl/src/main/java/pt/webdetails/cdf/dd/structure/DashboardStructure.java [hunk 2]
```diff
@@ -279,6 +279,24 @@ public void newfile( HashMap<String, Object> parameters ) throws Exception {
     }
   }
 
+    /**
+     * Deligates parameters setting to {@link #savesettings(HashMap)} adds updated wcdf to JSON object
+     * and returns it.
+     *
+     * @param parameters
```
Comment (afrjorge) [inline core/impl/src/main/java/pt/webdetails/cdf/dd/structure/DashboardStructure.java]: missing description

File: webclient/impls/pentaho/src/main/javascript/cdf-dd-base.js (hunk 1 above)
Comment (afrjorge) [reply webclient/impls/pentaho/src/main/javascript/cdf-dd-base.js]: We can change the callback function parameter name, but the object we receive will still be `{result: ..., status: ...}`. The `result.result` naming is used in many callback functions inside this file so maybe we can leave it like this for consistency.

File: core/impl/src/main/java/pt/webdetails/cdf/dd/structure/DashboardStructure.java [hunk 3]
```diff
@@ -279,6 +279,24 @@ public void newfile( HashMap<String, Object> parameters ) throws Exception {
...
+     * and returns it.
+     *
+     * @param parameters
+     * @return JSON object with wcdf that conatins updated parameters
+     * @throws Exception
+     */
+  public JSONObject saveSettingsToWcdf( HashMap<String, Object> parameters ) throws Exception {
+    savesettings( parameters );
+
+    JSONObject wcdfData = this.loadWcdfDescriptor( (String) parameters.get( "file" ) ).toJSON();
```
Comment (afrjorge) [inline core/impl/src/main/java/pt/webdetails/cdf/dd/structure/DashboardStructure.java]: Why not `final JSONObject wcdfData`? Or remove this local variable and pass the value in the `put` function.

**Comment by YuryBY** [comment]
@afrjorge, have updated. Thanks.

**Pull request merged**